pkg/kdump/kdump.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if test -f /proc/vmcore; then
+    #
+    # We are in dump capture kernel, nice.
+    #
+    MAX=5
+    DIR=/persist/kcrashes
+
+    # Create a folder
+    mkdir $DIR > /dev/null 2<&1
+
+    # Keep $MAX-1 fresh dumps
+    # shellcheck disable=SC2012
+    ls -t $DIR | tail -n +$MAX | xargs --no-run-if-empty -I '{}' rm $DIR/{}
+
+    # Get kernel panic from the dmesg of a crashed kernel
+    makedumpfile --dump-dmesg /proc/vmcore /tmp/dmesg > /dev/null
+    sed -n -e '/Kernel panic - not syncing/,$p' /tmp/dmesg > /tmp/backtrace
+
+    # Show backtrace from the dmesg of a crashed kernel
+    echo ">>>>>>>>>> Crashed kernel dmesg BEGIN <<<<<<<<<<" > /dev/kmsg
+    while read -r line; do echo "$line" > /dev/kmsg; done < /tmp/backtrace
+    echo ">>>>>>>>>> Crashed kernel dmesg END <<<<<<<<<<" > /dev/kmsg
+
+    # Collect a minimal kernel dump for security reasons
+    KDUMP_PATH="$DIR/kcrash-$(date +%Y-%m-%d-%H-%M-%S).dump"
+    makedumpfile -d 31 /proc/vmcore "$KDUMP_PATH" > /dev/null 2>&1
+    echo "kdump collected: $KDUMP_PATH" > /dev/kmsg
+
+    # Prepare reboot-reason and reboot-stack
+    echo "kernel panic, kdump collected: $KDUMP_PATH" > /persist/reboot-reason
+    cat /tmp/backtrace > /persist/reboot-stack
```

Comment:
Probably it would be great to store boot-reason here as well to properly handle by [GetBootReason](https://github.com/lf-edge/eve/blob/fb6b2a9b04a6fb79ee1ee92dc6071fcace1b0b59/pkg/pillar/agentlog/agentlog.go#L388-L393) after reboot, i.e. `echo "BootReasonKernel" > /persist/boot-reason`. Now I see empty `LastBootReason` field in device info message after manual kernel crash.

Reply:
Added, thanks